ModuleSystem/module_scripts_ai.py

```diff
@@ -2795,34 +2795,66 @@
 #White Mountains mountaintops, far apart, and update the values below with them
 tld_white_mountains_p1=( 26,-1)
 tld_white_mountains_p2=(-44,24)
+tld_white_mountains_min_y=-1200   # An Y*100 coordinate always North of the WM - I'm using Hornburg Y right now
 #And let me handle the rest obtaining the line equation y=mx+n
 tld_white_mountains_m=(tld_white_mountains_p1[1]-tld_white_mountains_p2[1])*1.0/(tld_white_mountains_p1[0]-tld_white_mountains_p2[0])
 tld_white_mountains_n=tld_white_mountains_p1[1]-tld_white_mountains_m*tld_white_mountains_p1[0]
 tld_white_mountains_m=int(round(tld_white_mountains_m*100,0))
 tld_white_mountains_n=int(round(tld_white_mountains_n*10000,0)) # When doing y=x*m + n, both x and m are mult * 100, k now must be mult an extra *100

```

Comment:
@ehsn  Okay, so according to https://www.mathsisfun.com/equation_of_line.html here the precomputed value `M` is the mountain-range _slope_ between those points you chose, and `N` is the vertical displacement from the origin ({0, 0}) (the '_intercept_') needed to position the reconstructed line.

Given those values we can compute/find the vertical Y position of any horizontal X point between the start and end of the White Mountains and compare it against the actual Y of the point to see if surpasses the line vertically or not.

    Y
    ^
    +---------------------+
    | -
    |     -
    |        [+] (height of the line at that horizontal position)
    |         |   -
    |         |       -
    +---------|-----------+ > X
    (0,0)    [*] (actual point position, south of both the line and the lowest part of the bounding box of the line)

Then there's an early bail out for the north, there's no point of doing expensive comparisons if the point is higher than the highest point of the line. Can't that be done for the lower part too?

I will take another look tomorrow, this is pretty interesting to look at. Maybe it needs some additional comments to demystify all this black speech of Mordor. What do you think? :)